localdev/seed/api.py

```diff
@@ -342,6 +344,10 @@ def _deserialize_courseware_object(self, serializer_cls, data):
         and creates it if it doesn't exist.
         """
         model_cls = serializer_cls.Meta.model
+        invalid_courseware_re = re.compile(
+            "course.*?\+[a-zA-Z0-9-]+?\+[a-zA-Z0-9-_]+\+"
+        )
+
```

Comment:
Since this is just a local dev command seeding sample data, I think we can live with not adding a regex. But, Even if we were to do that, there is a better place for this, Our [common openedx app ](https://github.com/mitodl/ol-django/blob/main/src/mitol/openedx/utils/courses.py) in Ol-Django.

We should move this validation from here to the app in ol-Django and then use it here and all the other places needed.

For the validation implementation, We can use similar logic as [this](https://github.com/mitodl/ol-django/blob/main/src/mitol/openedx/utils/courses.py#L21) with just a try-catch.